File: SDKs/Aspose.Cells_Cloud_SDK_For_Ruby/lib/aspose_cells_cloud/models/name.rb
module AsposeCellsCloud
  # 
  class Name < BaseObject
    attr_accessor :comment, :worksheet_index, :is_referred, :is_visible, :r1_c1_refers_to, :refers_to, :text, :link
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # 
        :'comment' => :'Comment',
        
        # 
        :'worksheet_index' => :'WorksheetIndex',
        
        # 
        :'is_referred' => :'IsReferred',
        
        # 
        :'is_visible' => :'IsVisible',
        
        # 
        :'r1_c1_refers_to' => :'R1C1RefersTo',
        
        # 
        :'refers_to' => :'RefersTo',
        
        # 
        :'text' => :'Text',
        
        # 
        :'link' => :'link'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'comment' => :'String',
        :'worksheet_index' => :'Integer',
        :'is_referred' => :'BOOLEAN',
        :'is_visible' => :'BOOLEAN',
        :'r1_c1_refers_to' => :'String',
        :'refers_to' => :'String',
        :'text' => :'String',
        :'link' => :'Link'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'Comment']
        self.comment = attributes[:'Comment']
      end
      
      if attributes[:'WorksheetIndex']
        self.worksheet_index = attributes[:'WorksheetIndex']
      end
      
      if attributes[:'IsReferred']
        self.is_referred = attributes[:'IsReferred']
      end
      
      if attributes[:'IsVisible']
        self.is_visible = attributes[:'IsVisible']
      end
      
      if attributes[:'R1C1RefersTo']
        self.r1_c1_refers_to = attributes[:'R1C1RefersTo']
      end
      
      if attributes[:'RefersTo']
        self.refers_to = attributes[:'RefersTo']
      end
      
      if attributes[:'Text']
        self.text = attributes[:'Text']
      end
      
      if attributes[:'link']
        self.link = attributes[:'link']
      end
      
    end

  end
end

File: SDKs/Aspose.Cells-Cloud-SDK-For-.NET/src/Com/Aspose/cells/Model/Name.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace Com.Aspose.Cells.Model {
  public class Name {
    public string Comment { get; set; }

    public int? WorksheetIndex { get; set; }

    public bool? IsReferred { get; set; }

    public bool? IsVisible { get; set; }

    public string R1C1RefersTo { get; set; }

    public string RefersTo { get; set; }

    public string Text { get; set; }

    public Link link { get; set; }

    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class Name {\n");
      sb.Append("  Comment: ").Append(Comment).Append("\n");
      sb.Append("  WorksheetIndex: ").Append(WorksheetIndex).Append("\n");
      sb.Append("  IsReferred: ").Append(IsReferred).Append("\n");
      sb.Append("  IsVisible: ").Append(IsVisible).Append("\n");
      sb.Append("  R1C1RefersTo: ").Append(R1C1RefersTo).Append("\n");
      sb.Append("  RefersTo: ").Append(RefersTo).Append("\n");
      sb.Append("  Text: ").Append(Text).Append("\n");
      sb.Append("  link: ").Append(link).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  }
